tests/mainnet/metaregistry/integrative/test_getters.py

```diff
@@ -0,0 +1,793 @@
+    if registry_id in [
+        METAREGISTRY_STABLE_REGISTRY_HANDLER_INDEX,
+        METAREGISTRY_STABLE_FACTORY_HANDLER_INDEX,
+    ]:
+        coins = registry.get_coins(pool)
+        balances = [0] * 8
+        for idx, coin in enumerate(coins):
+            if coin == brownie.ZERO_ADDRESS:
+                break
+            elif coin != brownie.ETH_ADDRESS:
+                balances[idx] = brownie.interface.ERC20(coin).balanceOf(pool)
+            else:
+                balances[idx] = curve_pool(pool).balance()
+
+        return balances
+
+    # admin balances for crypto swap is different:
+    v2_pool = curve_pool_v2(pool)
+    if registry_id == METAREGISTRY_CRYPTO_REGISTRY_HANDLER_INDEX:
+        fee_receiver = v2_pool.admin_fee_receiver()
+    elif registry_id == METAREGISTRY_CRYPTO_FACTORY_HANDLER_INDEX:
+        fee_receiver = registry.fee_receiver()
+
+    lp_token = brownie.interface.ERC20(metaregistry.get_lp_token(pool))
+    fee_receiver_token_balance_before = lp_token.balanceOf(fee_receiver)
+    v2_pool.claim_admin_fees({"from": alice})
+    claimed_lp_token_as_fee = lp_token.balanceOf(fee_receiver) - fee_receiver_token_balance_before
+    total_supply_lp_token = lp_token.totalSupply()
+
+    frac_admin_fee = int(claimed_lp_token_as_fee * 10**18 / total_supply_lp_token)
+
+    # get admin balances in individual assets:
+    reserves = metaregistry.get_balances(pool)
+    admin_balances = [0] * 8
+    for i in range(8):
+        admin_balances[i] = int(frac_admin_fee * reserves[i] / 10**18)
+
+    return admin_balances
+
+
+@pytest.mark.parametrize("pool_id", range(MAX_POOLS))
+def test_get_admin_balances(metaregistry, registry_pool_index_iterator, pool_id, alice, chain):
+
+    skip_if_pool_id_gt_max_pools_in_registry(pool_id, registry_pool_index_iterator)
+
+    registry_id, registry_handler, registry, pool = registry_pool_index_iterator[pool_id]
+
+    # we skip if pool was registered: this is because some pools are
+    # registered in the stable registry but are from the stable factory.
+    # skipping does not skip the test: since metaregistry chooses stable
+    # factory as the registry handler and not the registry in these cases
+    if check_pool_already_registered(metaregistry, pool, registry_handler):
+        pytest.skip()
+
+    # we skip if the pool is empty:
+    if sum(metaregistry.get_balances(pool)) == 0:
+        pytest.skip("empty pool: skipping.")
+
+    if brownie.interface.ERC20(metaregistry.get_lp_token(pool)).totalSupply() == 0:
+        pytest.skip("lp token supply is zero.")
+
+    chain.snapshot()
+    registry_reverts = False
+    try:
+        actual_output = _get_admin_balances_actuals(
+            registry_id, registry, pool, metaregistry, alice
+        )
+    except brownie.exceptions.VirtualMachineError:
+        registry_reverts = True
+    chain.revert()
+
+    if registry_reverts:
+        with brownie.reverts():
+            metaregistry.get_admin_balances(pool)
```

Comment:
can't the metaregistry return the right balances even if underlying registry reverts?

Reply:
i have now removed all revert handling